File: packages/android-sdk/src/main/kotlin/com/phosra/sdk/models/RuleCategory.kt
package com.phosra.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * All 45 rule categories matching the Go domain.RuleCategory constants.
 * Each enum entry's @SerialName matches the snake_case string used in JSON payloads.
 */
@Serializable
enum class RuleCategory {
    // ── Content rules ──────────────────────────────────────────────
    @SerialName("content_rating")
    CONTENT_RATING,

    @SerialName("content_block_title")
    CONTENT_BLOCK_TITLE,

    @SerialName("content_allow_title")
    CONTENT_ALLOW_TITLE,

    @SerialName("content_allowlist_mode")
    CONTENT_ALLOWLIST_MODE,

    @SerialName("content_descriptor_block")
    CONTENT_DESCRIPTOR_BLOCK,

    // ── Time rules ─────────────────────────────────────────────────
    @SerialName("time_daily_limit")
    TIME_DAILY_LIMIT,

    @SerialName("time_scheduled_hours")
    TIME_SCHEDULED_HOURS,

    @SerialName("time_per_app_limit")
    TIME_PER_APP_LIMIT,

    @SerialName("time_downtime")
    TIME_DOWNTIME,

    // ── Purchase rules ─────────────────────────────────────────────
    @SerialName("purchase_approval")
    PURCHASE_APPROVAL,

    @SerialName("purchase_spending_cap")
    PURCHASE_SPENDING_CAP,

    @SerialName("purchase_block_iap")
    PURCHASE_BLOCK_IAP,

    // ── Social rules ───────────────────────────────────────────────
    @SerialName("social_contacts")
    SOCIAL_CONTACTS,

    @SerialName("social_chat_control")
    SOCIAL_CHAT_CONTROL,

    @SerialName("social_multiplayer")
    SOCIAL_MULTIPLAYER,

    // ── Web rules ──────────────────────────────────────────────────
    @SerialName("web_safesearch")
    WEB_SAFESEARCH,

    @SerialName("web_category_block")
    WEB_CATEGORY_BLOCK,

    @SerialName("web_custom_allowlist")
    WEB_CUSTOM_ALLOWLIST,

    @SerialName("web_custom_blocklist")
    WEB_CUSTOM_BLOCKLIST,

    @SerialName("web_filter_level")
    WEB_FILTER_LEVEL,

    // ── Privacy rules ──────────────────────────────────────────────
    @SerialName("privacy_location")
    PRIVACY_LOCATION,

    @SerialName("privacy_profile_visibility")
    PRIVACY_PROFILE_VISIBILITY,

    @SerialName("privacy_data_sharing")
    PRIVACY_DATA_SHARING,

    @SerialName("privacy_account_creation")
    PRIVACY_ACCOUNT_CREATION,

    // ── Monitoring rules ───────────────────────────────────────────
    @SerialName("monitoring_activity")
    MONITORING_ACTIVITY,

    @SerialName("monitoring_alerts")
    MONITORING_ALERTS,

    // ── Algorithmic Safety rules (KOSA, KOSMA, CA SB 976, EU DSA) ──
    @SerialName("algo_feed_control")
    ALGO_FEED_CONTROL,

    @SerialName("addictive_design_control")
    ADDICTIVE_DESIGN_CONTROL,

    // ── Notification rules ─────────────────────────────────────────
    @SerialName("notification_curfew")
    NOTIFICATION_CURFEW,

    @SerialName("usage_timer_notification")
    USAGE_TIMER_NOTIFICATION,

    // ── Advertising & Data rules (COPPA 2.0, EU DSA, India DPDPA) ──
    @SerialName("targeted_ad_block")
    TARGETED_AD_BLOCK,

    @SerialName("dm_restriction")
    DM_RESTRICTION,

    @SerialName("age_gate")
    AGE_GATE,

    @SerialName("data_deletion_request")
    DATA_DELETION_REQUEST,

    @SerialName("geolocation_opt_in")
    GEOLOCATION_OPT_IN,

    // ── Compliance expansion rules ─────────────────────────────────
    @SerialName("csam_reporting")
    CSAM_REPORTING,

    @SerialName("library_filter_compliance")
    LIBRARY_FILTER_COMPLIANCE,

    @SerialName("ai_minor_interaction")
    AI_MINOR_INTERACTION,

    @SerialName("social_media_min_age")
    SOCIAL_MEDIA_MIN_AGE,

    @SerialName("image_rights_minor")
    IMAGE_RIGHTS_MINOR,

    // ── Legislation-driven expansion (2025) ────────────────────────
    @SerialName("parental_consent_gate")
    PARENTAL_CONSENT_GATE,

    @SerialName("parental_event_notification")
    PARENTAL_EVENT_NOTIFICATION,

    @SerialName("screen_time_report")
    SCREEN_TIME_REPORT,

    @SerialName("commercial_data_ban")
    COMMERCIAL_DATA_BAN,

    @SerialName("algorithmic_audit")
    ALGORITHMIC_AUDIT;

    companion object {
        /** Returns all 45 rule categories. */
        fun all(): List<RuleCategory> = entries.toList()
    }
}

File: packages/ios-sdk/Sources/PhosraSDK/Models/RuleCategory.swift
import Foundation

/// All 45 Phosra rule categories, matching the Go `domain.RuleCategory` constants.
///
/// Each case maps to its snake_case string value used in the JSON API.
public enum RuleCategory: String, Codable, CaseIterable, Sendable {
    // MARK: - Content Rules
    case contentRating = "content_rating"
    case contentBlockTitle = "content_block_title"
    case contentAllowTitle = "content_allow_title"
    case contentAllowlistMode = "content_allowlist_mode"
    case contentDescriptorBlock = "content_descriptor_block"

    // MARK: - Time Rules
    case timeDailyLimit = "time_daily_limit"
    case timeScheduledHours = "time_scheduled_hours"
    case timePerAppLimit = "time_per_app_limit"
    case timeDowntime = "time_downtime"

    // MARK: - Purchase Rules
    case purchaseApproval = "purchase_approval"
    case purchaseSpendingCap = "purchase_spending_cap"
    case purchaseBlockIAP = "purchase_block_iap"

    // MARK: - Social Rules
    case socialContacts = "social_contacts"
    case socialChatControl = "social_chat_control"
    case socialMultiplayer = "social_multiplayer"

    // MARK: - Web Rules
    case webSafesearch = "web_safesearch"
    case webCategoryBlock = "web_category_block"
    case webCustomAllowlist = "web_custom_allowlist"
    case webCustomBlocklist = "web_custom_blocklist"
    case webFilterLevel = "web_filter_level"

    // MARK: - Privacy Rules
    case privacyLocation = "privacy_location"
    case privacyProfileVisibility = "privacy_profile_visibility"
    case privacyDataSharing = "privacy_data_sharing"
    case privacyAccountCreation = "privacy_account_creation"

    // MARK: - Monitoring Rules
    case monitoringActivity = "monitoring_activity"
    case monitoringAlerts = "monitoring_alerts"

    // MARK: - Algorithmic Safety Rules (KOSA, KOSMA, CA SB 976, EU DSA)
    case algoFeedControl = "algo_feed_control"
    case addictiveDesignControl = "addictive_design_control"

    // MARK: - Notification Rules (VA SB 854, NY SAFE for Kids, MN HF 2, TN HB 1891)
    case notificationCurfew = "notification_curfew"
    case usageTimerNotification = "usage_timer_notification"

    // MARK: - Advertising & Data Rules (COPPA 2.0, EU DSA, India DPDPA, CT SB 3, MD Kids Code)
    case targetedAdBlock = "targeted_ad_block"
    case dmRestriction = "dm_restriction"
    case ageGate = "age_gate"
    case dataDeletionRequest = "data_deletion_request"
    case geolocationOptIn = "geolocation_opt_in"

    // MARK: - Compliance Expansion Rules
    case csamReporting = "csam_reporting"
    case libraryFilterCompliance = "library_filter_compliance"
    case aiMinorInteraction = "ai_minor_interaction"
    case socialMediaMinAge = "social_media_min_age"
    case imageRightsMinor = "image_rights_minor"

    // MARK: - Legislation-Driven Expansion (2025)
    case parentalConsentGate = "parental_consent_gate"
    case parentalEventNotification = "parental_event_notification"
    case screenTimeReport = "screen_time_report"
    case commercialDataBan = "commercial_data_ban"
    case algorithmicAudit = "algorithmic_audit"
}
